pfcp/handler.HandlePfcpAssociationSetupRequest: nil pointer dereference

I'm testing free5GC SMF with OAI-CN5G-UPF, and I encountered free5GC SMF crash.

Packet trace: [pfcp-upipri.pcapng.gz](https://github.com/user-attachments/files/18116894/pfcp-upipri.pcapng.gz)
The offending packet is transmitted by OAI-CN5G-UPF and received by free5GC SMF.
```text
Packet Forwarding Control Protocol
    Flags: 0x20
    Message Type: PFCP Association Setup Request (5)
    Length: 31
    Sequence Number: 11639750
    Spare: 0
    Node ID : IPv4 address: 172.25.196.5
    Recovery Time Stamp : Dec 12, 2024 02:25:51.000000000 UTC
    UP Function Features : 
```

It causes free5GC SMF to crash with the following log message:
```text
2024-12-12T19:57:35.533847669Z[36m [INFO][SMF][PFCP] [0mHandle PFCP Association Setup Request with NodeID[172.25.196.5]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb328d6]

goroutine 7 [running]:
github.com/free5gc/smf/internal/pfcp/handler.HandlePfcpAssociationSetupRequest(0xc000524000)
	/go/src/free5gc/NFs/smf/internal/pfcp/handler/handler.go:42 +0x196
github.com/free5gc/smf/internal/pfcp.Dispatch(0x0?)
	/go/src/free5gc/NFs/smf/internal/pfcp/dispatcher.go:17 +0xd9
created by github.com/free5gc/smf/internal/pfcp/udp.Run.func2 in goroutine 26
	/go/src/free5gc/NFs/smf/internal/pfcp/udp/udp.go:71 +0x127
```

The code in the stacktrace is:
https://github.com/free5gc/smf/blob/62925c4a5840f118b42f245b5fef492898f9f0bd/internal/pfcp/handler/handler.go#L42
It seems that `req.UserPlaneIPResourceInformation` is being dereferenced without checking it's non-nil.

According to [TS 129 244 V15](https://www.etsi.org/deliver/etsi_ts/129200_129299/129244/15.08.00_60/ts_129244v150800p.pdf) section 7.4.4.1, the User Plane IP Resource Information IE is optional, so that SMF cannot depend it's present.
Moreover, this IE has been deleted in [TS 129 244 V16](https://www.etsi.org/deliver/etsi_ts/129200_129299/129244/16.06.00_60/ts_129244v160600p.pdf) (section 8.2.82 is "void"), which signifies that it is unimportant and shouldn't be relied upon.

--- a/internal/context/upf.go
+++ b/internal/context/upf.go
@@ -68,7 +68,6 @@ type UPF struct {
 	uuid              uuid.UUID
 	NodeID            pfcpType.NodeID
 	Addr              string
-	UPIPInfo          pfcpType.UserPlaneIPResourceInformation
 	RecoveryTimeStamp time.Time
 
 	AssociationContext context.Context
@@ -351,19 +350,6 @@ func RemoveUPFNodeByNodeID(nodeID pfcpType.NodeID) bool {
 		return true
 	}
 	return false
-}
-
-func SelectUPFByDnn(dnn string) *UPF {
-	var upf *UPF
-	upfPool.Range(func(key, value interface{}) bool {
-		upf = value.(*UPF)
-		if upf.UPIPInfo.Assoni && upf.UPIPInfo.NetworkInstance.NetworkInstance == dnn {
-			return false
-		}
-		upf = nil
-		return true
-	})
-	return upf
 }
 
 func (upf *UPF) GetUPFIP() string {

--- a/internal/pfcp/handler/handler.go
+++ b/internal/pfcp/handler/handler.go
@@ -39,8 +39,6 @@ func HandlePfcpAssociationSetupRequest(msg *pfcpUdp.Message) {
 		return
 	}
 
-	upf.UPIPInfo = *req.UserPlaneIPResourceInformation
-
 	// Response with PFCP Association Setup Response
 	cause := pfcpType.Cause{
 		CauseValue: pfcpType.CauseRequestAccepted,

--- a/internal/sbi/processor/association.go
+++ b/internal/sbi/processor/association.go
@@ -104,13 +104,7 @@ func setupPfcpAssociation(upf *smf_context.UPF, upfStr string) error {
 	}
 
 	logger.MainLog.Infof("Received PFCP Association Setup Accepted Response from UPF%s", upfStr)
-
-	if rsp.UserPlaneIPResourceInformation != nil {
-		upf.UPIPInfo = *rsp.UserPlaneIPResourceInformation
-
-		logger.MainLog.Infof("UPF(%s)[%s] setup association",
-			upf.NodeID.ResolveNodeIdToIp().String(), upf.UPIPInfo.NetworkInstance.NetworkInstance)
-	}
+	logger.MainLog.Infof("UPF(%s) setup association", upf.NodeID.ResolveNodeIdToIp().String())
 
 	return nil
 }

